src/commands/push.js

```diff
@@ -59,6 +60,12 @@ export async function push ({
     } else {
       fullRef = await GitRefManager.expand({ fs, gitdir, ref })
     }
+    let fullRemoteRef
+    if (!remoteRef) {
+      fullRemoteRef = fullRef
+    } else {
+      fullRemoteRef = await GitRefManager.expand({ fs, gitdir, ref: remoteRef })
```

Comment:
Technically this may not do what we want, because `.expand` searches through _local_ refs. I think I see how to fix this though.

Reply:
Ah. Yeah, actually this will crash it if trying to push to a new branch (that doesn't exist yet). I got this.